### ISSUE TITLE
[Phase 1] Data Infrastructure - Loading and Validation

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/particle-collision-classifier#2]
## Objective

Build the data loading infrastructure for handling large-scale HEP datasets (10M+ events) with memory-efficient operations.

**Part of:** #1

## Dependencies

None (first phase)

### Data Loader Implementation
- [ ] Create `src/data/loader.py` with `HEPDataLoader` class
- [ ] Implement chunked CSV reading for memory efficiency
- [ ] Add stratified train/val/test splitting
- [ ] Support all three datasets: HIGGS, HEPMASS, SUSY

### Data Downloader
- [ ] Create `src/data/downloader.py` with download utilities
- [ ] Implement retry logic for large file downloads
- [ ] Add progress bar with tqdm

### Data Validator
- [ ] Create `src/data/validator.py` with validation checks
- [ ] Validate column counts and data types
- [ ] Check for missing values and corrupted rows

### Configuration
- [ ] Create `configs/datasets.yaml` with dataset metadata

### Tests
- [ ] Write unit tests in `tests/unit/test_loader.py`
- [ ] Achieve >80% coverage on data module

### HEPDataLoader Skeleton

```python
# src/data/loader.py
import os
from typing import Optional, Tuple
import pandas as pd
from sklearn.model_selection import train_test_split

DATASET_CONFIG = {
    "higgs": {
        "url": "https://archive.ics.uci.edu/ml/machine-learning-databases/00280/HIGGS.csv.gz",
        "filename": "HIGGS.csv.gz",
        "n_features": 28,
        "label_col": 0,
    },
    "hepmass": {
        "url": "https://archive.ics.uci.edu/ml/machine-learning-databases/00347/all_train.csv.gz",
        "filename": "hepmass_train.csv.gz",
        "n_features": 27,
        "label_col": 0,
    },
    "susy": {
        "url": "https://archive.ics.uci.edu/ml/machine-learning-databases/00279/SUSY.csv.gz",
        "filename": "SUSY.csv.gz",
        "n_features": 18,
        "label_col": 0,
    },
}

class HEPDataLoader:
    def __init__(
        self,
        dataset: str,
        data_dir: str = "data/raw",
        sample_size: Optional[int] = None,
        random_state: int = 42
    ):
        if dataset.lower() not in DATASET_CONFIG:
            raise ValueError(f"Unknown dataset: {dataset}")
        self.dataset = dataset.lower()
        self.config = DATASET_CONFIG[self.dataset]
        self.data_dir = data_dir
        self.sample_size = sample_size
        self.random_state = random_state

    def load(self) -> Tuple[pd.DataFrame, pd.Series]:
        # Implementation here
        pass

    def load_split(self, test_size=0.15, val_size=0.15):
        # Implementation here
        pass
```

### First Test

```python
# tests/unit/test_loader.py
import pytest
from src.data.loader import HEPDataLoader

def test_loader_raises_on_unknown_dataset():
    with pytest.raises(ValueError):
        HEPDataLoader("nonexistent")

def test_loader_config_correct():
    loader = HEPDataLoader("higgs")
    assert loader.config["n_features"] == 28
```

## Definition of Done

- [ ] Can load 100K samples in <10 seconds
- [ ] All three datasets supported
- [ ] Unit tests pass with >80% coverage
- [ ] Data validation catches common issues

## Technical Notes

- HIGGS and SUSY use first column as label
- All datasets are gzip compressed
- Use `pd.read_csv(..., compression='gzip')` for direct reading
